src/PrestaShopBundle/Controller/Admin/Configure/ShopParameters/MaintenanceController.php

```diff
@@ -0,0 +1,114 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ * 
+ * DISCLAIMER
+ * 
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ * 
+ *  @author PrestaShop SA <contact@prestashop.com>
+ *  @copyright  2007-2017 PrestaShop SA
+ *  @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *  International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\Admin\Configure\ShopParameters;
+
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Security\Voter\PageVoter;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Tools;
+
+/**
+ * Responsible of "Configure > Shop Parameters > General > Maintenance" page
+ */
+class MaintenanceController extends FrameworkBundleAdminController
+{
+    const CONTROLLER_NAME = 'AdminMaintenance';
+
+    /**
+     * @var FormInterface
+     * @return Response
+     */
+    public function indexAction(Request $request, FormInterface $form = null)
+    {
+        if (is_null($form)) {
+            $form = $this->get('prestashop.adapter.maintenance.form_handler')->getForm();
+        }
+
+        $twigValues = array(
+            'layoutHeaderToolbarBtn' => array(),
+            'layoutTitle' => $this->get('translator')->trans('Maintenance', array(), 'Admin.Navigation.Menu'),
+            'requireAddonsSearch' => true,
+            'requireBulkActions' => false,
+            'showContentHeader' => true,
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink('AdminMaintenance'),
+            'requireFilterStatus' => false,
+            'form' => $form->createView(),
+            'currentIp' => $request->getClientIp(),
+        );
+
+        return $this->render('@ShopParameters/maintenance.html.twig', $twigValues);
+    }
+
+    /**
+     * @return RedirectResponse
+     */
+    public function processFormAction(Request $request)
+    {
+        $redirectResponse = $this->redirectToRoute('admin_maintenance');
+        if ($this->isDemoModeEnabled()) {
+            $this->addFlash('error', $this->getDemoModeErrorMessage());
+
+            return $redirectResponse;
+        }
+
+        if (!in_array(
+            $this->authorizationLevel($this::CONTROLLER_NAME),
+            array(
+                PageVoter::LEVEL_READ,
+                PageVoter::LEVEL_UPDATE,
+                PageVoter::LEVEL_CREATE,
+                PageVoter::LEVEL_DELETE,
+            )
+        )) {
+            $this->addFlash('error', $this->trans('You do not have permission to update this.', 'Admin.Notifications.Error'));
```

Comment:
Okay, this string appears in the Crowdin but in the "/1.7.3/messages.xlf" folder, which is the catch-all folder for strings with no domains assigned to yet. And it should be in Admin.Notifications.Error, you're right. Let's merge this PR with these right domains and see what happen.

Reply:
As far as I know, this will create a duplicate in `messages` & `Admin.Notifications.Error`.
We should remove remaining strings in `messages`, so I agree with you.

Reply:
Yeah !